sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/codegen/UnsafeRowWriter.java

```diff
@@ -95,41 +99,75 @@ public void alignToWords(int numBytes) {
     }
   }
 
-  public void writeCompactDecimal(int ordinal, Decimal input, int precision, int scale) {
-    // make sure Decimal object has the same scale as DecimalType
-    if (input.changePrecision(precision, scale)) {
-      Platform.putLong(holder.buffer, getFieldOffset(ordinal), input.toUnscaledLong());
-    } else {
-      setNullAt(ordinal);
-    }
+  public void write(int ordinal, boolean value) {
+    Platform.putBoolean(holder.buffer, getFieldOffset(ordinal), value);
```

Comment:
I think we should zero-out the whole 8 bytes before writing.
For inner unsafe row, the start position of it may change during mulitly wirtes. So these 8 bytes are not always clean, if we only write a boolean here, the remaing 7 bytes maybe not zero, and will break the equality of unsafe row.